composer/core/evaluator.py

```diff
@@ -39,6 +41,23 @@ def should_eval(state: State, event: Event):
         if int(eval_interval) <= 0:
             return False
 
+        # if requested, evaluate at the end of training, as long as the length of training is specified.
+        if eval_at_fit_end and state.max_duration is not None:
```

Comment:
Is it possible to use `Event.FIT_END` to simplify the logic?

Reply:
> Is it possible to use Event.FIT_END to simplify the logic?

Let me try, I initially had an implementation that checked `if state.get_elapsed_duration==1.0` , but the following scenario happens. Imagine a case where `eval_interval=1ba` and `max_duration=Xep`, where X is an integer. Let's say that there is 5 batches in an epoch.

```
*model performs gradient updates*
should_eval runs: Time: 2ep, 11ba
*model performs gradient updates*
should_eval runs: Time: 2ep, 12ba
*model performs gradient updates*
should_eval runs: Time: 2ep, 13ba
*model performs gradient updates*
should_eval runs: Time: 2ep, 14ba
*model performs gradient updates*
should_eval runs: Time: 2ep, 15ba <--- A
should_eval runs: Time: 3ep, 15ba <--- B
```

Since there are no gradient updates between event A and B, we shouldn't duplicate evaluation, and should only run it once. I assume `Event.FIT_END` is called after event B, and in this scenario, my current impression is that would lead to running evaluation twice -- this is why I added logic to identify the last _batch_ trained if the `eval_interval` is batches (this won't be a problem if `eval_interval` is epochs). 

Does that make sense? Do you see a way to sidestep this problem and still use `Event.FIT_END`?

Reply:
This is done!